prowler/lib/outputs/common.py

```diff
@@ -22,78 +23,93 @@ def get_provider_data_mapping(provider) -> dict:
     return data
 
 
-def generate_provider_output(provider, finding, csv_data) -> FindingOutput:
+def generate_output(provider, finding, output_options) -> Finding:
+    provider_data_mapping = get_provider_data_mapping(provider)
+    common_finding_data = fill_common_finding_data(
+        finding, output_options.unix_timestamp
+    )
+    output_data = {}
+    output_data.update(provider_data_mapping)
+    output_data.update(common_finding_data)
+    output_data["compliance"] = get_check_compliance(
+        finding, provider.type, output_options
+    )
+    finding_output = generate_provider_output(provider, finding, output_data)
+    return finding_output
+
+
+def generate_provider_output(provider, finding, output_data) -> Finding:
```

Comment:
What if we move this into the `Finding` class?

Please also add tests and a docsting to the function to explain inputs, outputs, and what it does.

Reply:
Done